src/controllers/bookController.ts

```diff
@@ -1,29 +1,151 @@
+            },
+        },
+    };
+
     constructor() {
         this.router = Router();
         this.router.get('/:id', this.getBook.bind(this));
+        this.router.get('/', this.getBooks.bind(this));
+        this.router.post(
+            '/:bookTitle/:copies/:ISBN/:authorFirst/:authorLast',
+            this.createBook.bind(this),
+        );
+    }
+
+    async getBook(req: Request, res: Response) {
+        const connection = new Connection(this.config);
+        let requestStatement = new String(); //Plz ignore
+        const output = [];
 
-        this.router.post('/', this.createBook.bind(this));
+        // Setup event handler when the connection is established.
+        connection.on('connect', async function (err) {
+            if (err) {
+                console.log('Error: ', err);
+                throw err;
+            }
+            let id = Number(req.params.id);
+            if (isNaN(id)) {
+                return res.status(400).json({ status: 'Invalid Id' });
+            } else {
+                const sqlRequest = `SELECT * FROM books WHERE id = ${id}`;
+                const databaseRequest = new TediousRequest(
+                    sqlRequest,
+                    function (err, _rowCount, rows) {
+                        if (err) {
+                            console.log(err);
+                            throw err;
+                        } else {
+                            if (rows.length == 0) {
+                                return res
+                                    .status(400)
+                                    .json({ status: 'book not found' });
+                            } else {
+                                for (let j = 0; j < rows[0].length; j++) {
+                                    output.push(rows[0][j].value);
+                                }
+                                return res.status(200).json(output);
+                            }
+                        }
+                    },
+                );
+                requestStatement = connection.execSql(databaseRequest);
+            }
+        });
+        connection.connect();
     }
 
-    getBook(req: Request, res: Response) {
-        // TODO: implement functionality
-        return res.status(500).json({
-            error: 'server_error',
-            error_description: 'Endpoint not implemented yet.',
+    async createBook(req: Request, res: Response) {
+        const connection = new Connection(this.config);
+        let requestStatement = new String(); //Plz ignore
+        const output = [];
+
+        // Setup event handler when the connection is established.
+        connection.on('connect', async function (err) {
+            if (err) {
+                console.log('Error: ', err);
+                throw err;
+            }
+            const parsedBookTitle = req.params.bookTitle.split('_').join(' ');
+
+            const sqlRequest = `INSERT INTO books (BookTitle, Copies, ISBN) 
```

Comment:
I know I spoke about this yesterday, but this is very dangerous!! A big open door for SQL injection and all of the issues that come with that.

It's good to get into the habit of using placeholders and the library methods and validation to insert them when working with SQL to avoid things like this.